homeassistant/components/unifi/sensor.py

```diff
@@ -192,6 +199,88 @@ def async_device_state_value_fn(hub: UnifiHub, device: Device) -> str:
     return DEVICE_STATES[device.state]
 
 
+@callback
+def async_device_wan_monitor_latency_supported_fn(
```

Comment:
This method is the one which should be the guard for existing values for the entity. If this passes and the entity is loaded then value_fn should be able to expect the value to exist.

This is also supported in the UnifiEntity.async_signalling_callback method which always uses supported_fn first before the value_fn. Check it out

So I would like to see supported_fn and value_fn to not depend on each other. Please make an external method if you want shared logic.

Reply:
Something like in my latest commit?